module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -131,66 +233,108 @@ namespace module::localisation {
+                                            (fd.dimensions.field_width / 2.0),
+                                            M_PI_2);
+
+            // // Left side goal line
+            // config.start_state.emplace_back(
+            //     (-fd.dimensions.field_length / 2.0),
+            //     (-fd.dimensions.field_width / 2.0) + ((fd.dimensions.field_width - fd.dimensions.goal_width) / 4.0),
+            //     0.0);
+
+            // // Right side goal line
+            // config.start_state.emplace_back(
+            //     (-fd.dimensions.field_length / 2.0),
+            //     (fd.dimensions.field_width / 2.0) - ((fd.dimensions.field_width - fd.dimensions.goal_width) / 4.0),
+            //     0.0);
+
+            // Penalty shootout position
+            // config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) +
+            // fd.dimensions.penalty_mark_distance,
+            //                                 0.0,
+            //                                 -M_PI);
+
+            filter.set_state(config.start_state,
+                             std::vector<Eigen::Vector3d>(config.start_state.size(), config.start_variance));
         });
     }
 
+
+    // True goal posiiton
     Eigen::Vector3d RobotParticleLocalisation::getFieldPosition(const VisionGoal& goal,
-                                                                const message::support::FieldDescription& fd) const {
+                                                                const message::support::FieldDescription& fd,
+                                                                const bool isOwn) const {
         Eigen::Vector3d position;
 
-        const bool left  = (goal.side != VisionGoal::Side::RIGHT);
-        const bool right = (goal.side != VisionGoal::Side::LEFT);
-        const bool own   = (goal.team != VisionGoal::Team::OPPONENT);
-        const bool opp   = (goal.team != VisionGoal::Team::OWN);
+        const bool left  = (goal.side == VisionGoal::Side::LEFT);
+        const bool right = (goal.side == VisionGoal::Side::RIGHT);
+
+        // TODO This should be removed from the message
+        // const bool own   = (goal.team != VisionGoal::Team::OPPONENT);
+        // const bool opp   = (goal.team != VisionGoal::Team::OWN);
 
-        if (own && left) {
+        if (isOwn && left) {
             position = Eigen::Vector3d(fd.goalpost_own_l.x(), fd.goalpost_own_l.y(), 0);
         }
-        if (own && right) {
+        if (isOwn && right) {
             position = Eigen::Vector3d(fd.goalpost_own_r.x(), fd.goalpost_own_r.y(), 0);
         }
-        if (opp && left) {
+        if (!isOwn && left) {
             position = Eigen::Vector3d(fd.goalpost_opp_l.x(), fd.goalpost_opp_l.y(), 0);
         }
-        if (opp && right) {
+        if (!isOwn && right) {
             position = Eigen::Vector3d(fd.goalpost_opp_r.x(), fd.goalpost_opp_r.y(), 0);
         }
 
         return position;
```

Comment:
so we return an uninitialised vector if `goal.size == VisionGoal::Side::UNKNOWN_SIDE`?????????
pls fix

Reply:
This function is only for getting truth data, so in theory we will never be asking it about unknown sides.